### ISSUE TITLE
Object stops despite bounce set to 1 when Continuous CD is on

### ISSUE THREAD

**Issue by Regrad** [godotengine/godot#79334]
### Godot version

4.1

### System information

Windows 11

### Issue description

When Continuous CD is on, object stops despite bounce 1:
https://youtu.be/bsOe_VxWv2M

### Steps to reproduce

Open project

### Minimal reproduction project

[Godot_continuous_cd.zip](https://github.com/godotengine/godot/files/12014038/Godot_continuous_cd.zip)

**Comment by quinnyo** [comment]
Couldn't see an issue in the MRP. The balls bounce and never stop. Without CCD, the balls go through the (thin) wall.

**Comment by Regrad** [comment]
After hitting the ball against the wall, it loses all speed and does not bounce.
https://drive.google.com/file/d/1D4BywwfueCxzpjoKjTiGctCnvawV8Uhr/view?usp=sharing

**Comment by ohthesetrees** [comment]
I actually ran into this myself. I've got a small repro project here: https://www.dropbox.com/scl/fi/ulr35fwrx8czgqhhpawgd/break-it.zip?rlkey=r8r5bme76g6y33g8aaa21lh6b&dl=0. Press space to launch the ball and notice the velocity changes on bounce despite the bounce value being set to 1. I've also toggled off CCD in which case the velocity is maintained but the ball can penetrate through the walls.

I'm going to try digging into the code myself but I am not sure what I'm looking for so no guarantees I can actually find a fix for it.
